packages/enzyme-adapter-react-16/src/detectFiberTags.jsx

```diff
@@ -38,6 +67,9 @@ module.exports = function detectFiberTags() {
     // eslint-disable-next-line no-unused-vars
     FwdRef = React.forwardRef((props, ref) => null);
   }
+  if (supportsLazy) {
+    LazyComponent = React.lazy(() => import('./_helpers/dynamicImportedComponent'));
```

Comment:
There's zero point in using `import()` here - it has to be transpiled, and it's never safe to transpile it below the app level, so this would have to go untranspiled, and that'd be a breaking change.

It'd be much simpler to use `() => Promise.resolve().then(() => ({ default: require(…) }))`.

Separately, I'm not sure why this can't be even simpler - `React.lazy(() => Promise.resolve().then(() => ({ default() { return null; } }))`

Reply:
Ah... yeah, the `Promise.resolve().then(() => ({ default() { return null; } }))` indeed works. If so I think all the added eslint plugins for `import` syntax and `babel-eslint` can be removed. Thanks for this solution!

Reply:
For what it's worth, this way is *much* cleaner in the sense that it preserves a single source of truth of "how `import()` should work in node" - but due to the underspecified nature of ESM, until node ships unflagged ESM, it's not safe for anything but the top-level app to define "how `import()` works", unfortunately.

Reply:
I think maybe it's worth to add a utility to make it clearer like

```js
const fakeDynamicImport = (module) => Promise.resolve().then(() => ({ default: module}))

// so we can write
const LazyComponent = React.lazy(fakeDynamicImport(SomeComponent))
```
for all the places using `React.lazy` in enzyme packages. But I'm not sure where to put this utility.

Reply:
Sounds great! If it'd be used by only adapters, it should be in adapter-utils; if by enzyme as well, then probably in enzyme itself; since it's likely to only be used in the 16 adapter for now but the 17 adapter later, probably in adapter-utils?

Reply:
`React.lazy` only appears in 16 adapter and tests. Looks like adapter-utils is a good place.